<?xml version="1.0"?>
<launch>

  <arg name="fps" default="20"/>
  <arg name="CAM" default="3"/>
  <arg name="trigger_mode" default="0"/>
  <arg name="ptp_timestamp" default="0"/>
  <arg name="camera_parameter_ns" default="config"/>


  <node name="image_proc" pkg="image_proc" type="image_proc" ns="genicam_cam$(arg CAM)"/>

  <node type="genicam_node" name="genicam_cam$(arg CAM)" pkg="genicam_cvb" respawn="true" respawn_delay="10" output="screen">
    <param name="capability_group" value="genicam_cam$(arg CAM)/"/>
    <param name="respawn/max" value="10" />
    <param name="respawn/min_runtime" value="60" />
    <param name="respawn/delay" value="2" />
    <param name="driverpath" value="$(find genicam_cvb)/driver/SP-5000M-GE2_$(arg CAM).ini"/>
    <param name="ros_calibration" value="file://$(find genicam_cvb)/calib/jai_5000_m.yaml"/>
    <param name="expected_fps" value="$(arg fps)"/>
    <param name="print_fps" value="1"/>
    <param name="stream_counter_rst" value="1"/>
    <param name="topic" value="image_raw"/>
    <param name="frame" value="cam$(arg CAM)"/>
    <param name="gstreamer_sink_for_storage" value="shmsink socket-path=/tmp/cam/jai_$(arg CAM)_store sync=true wait-for-connection=false shm-size=10000000"/>
    <param name="color_channels_src" value="1"/>
    <param name="color_channels_dst" value="1"/>
    <param name="mono_as_bayer" value="0"/>
    <param name="trigger_mode" value="$(arg trigger_mode)"/>
    <param name="ptp_timestamp" value="$(arg ptp_timestamp)"/>
    <param name="camera_parameter_ns" value="$(arg camera_parameter_ns)"/>
    <rosparam ns="$(arg camera_parameter_ns)">
      Width: '2560'
      Height: '2048'
      PixelFormat: '17301505'
      GainSelector: '1'
      Gain: '1'
      ExposureMode: '1'
      ExposureTime: '17000'
      HighDynamicRangeMode: '0'
      HighDynamicRangeSlope: '2'
      GevIEEE1588: '1'
    </rosparam>
  </node>

</launch>
